frontend/src/api/actions/positions.js

```diff
@@ -58,12 +59,28 @@ const instructorsToInstructorIds = flattenIdFactory(
     "instructor_ids",
     true
 );
-const contractTemplateToInstructorIds = flattenIdFactory(
+const contractTemplateToContractTemplateId = flattenIdFactory(
     "contract_template",
     "contract_template_id"
 );
+
+const instructorToInstructorId = flattenIdFactory(
+    "instructor",
+    "instructor_id"
+);
+
+const applicantToApplicantId = flattenIdFactory("applicant", "applicant_id");
+
 function prepForApi(data) {
-    return contractTemplateToInstructorIds(instructorsToInstructorIds(data));
+    data.instructor_preferences = (
```

Comment:
Don't mutate the incoming data...

If `instructor_preferences` is null, we want to leave it as null, lest we accidentally delete the instructor preferences by sending a blank array.